src/harness/sas.py

```diff
@@ -287,6 +289,14 @@ def InjectSasImplementationRecord(self, request):
   def InjectEscSensorDataRecord(self, request):
     _RequestPost('https://%s/admin/injectdata/esc_sensor' % self._base_url, request,
                  self._tls_config)
+ 
+  def TriggerCreateActivityDump(self, request):
```

Comment:
I used the extra.config for each test case which is to provide the extra data for the required test case. Because the device_a, device_b, device_c, device_d... missing some fields or in the future we will need to change the fields value for example: longitude and latitude may need to be changed in the future for different test cases. So I used the extra.config to provide the extra data fields for future configuration of each individual test case.

Reply:
The other way I can do is to create extra device.json in test data folder for the specific test case, I think it is the same as providing them together in extra.config. All the data from extra.config will be loaded into default.config. In the test case the data will be loaded from default.config

Reply:
I think there's a misunderstanding here about how the configurations are supposed to work. There should be *one* configuration file which is used as the input to the test case (e.g. FAD.2). That file should contain at least the following things:
 - SAS TH base URL
 - SAS TH cert/key
 - SAS TH ESC file contents
 - SAS TH PPA file contents
 - SAS TH CBSD/Grant file contents
 - Registration request for C2
 - Grant request for C2
 - Registration request for C4
 - Grant request for C4

The test case should also generate a default config if there's no existing config. In this case you might want to e.g. load device_a, alter its parameters slightly, and use it as the registration request for C2.

ITS is NOT expected to change device_a.json, etc. They are expect to create their own config file for each instance of the test case that they wish to execute. Perhaps you can sync with Janet at Federated Wireless for additional details and conventions.